dace/codegen/targets/cuda.py

```diff
@@ -426,25 +427,26 @@ def allocate_array(self, sdfg, dfg, state_id, node, nodedesc, function_stream, d
     def allocate_stream(self, sdfg, dfg, state_id, node, nodedesc, function_stream, declaration_stream,
                         allocation_stream):
         dataname = node.data
-        allocname = cpp.ptr(dataname, nodedesc, sdfg)
+        allocname = cpp.ptr(dataname, nodedesc, sdfg, self._frame)
         if nodedesc.storage == dtypes.StorageType.GPU_Global:
             fmtargs = {
-                'name': dataname,
+                'name': allocname,  # TODO: Handle persistent streams
```

Comment:
So is `name` obsolete?

Reply:
no, we need to bring it back later because of a `__state->` opener that we don't need in the declaration stream. However, I don't know how to take care of that right now. Hence the TODO